torch/_inductor/lowering.py

```diff
@@ -3272,18 +3272,23 @@ def scatter_reduce_(self, dim: int, index, src, reduce, *, include_self: bool =
         len(aten.scatter_reduce_.overloads()) == 1
         and "two" in aten.scatter_reduce_.overloads()
     ), "aten.scatter_reduce_.two is not the unique overload of aten.scatter_reduce_"
-    fallback_result = scatter_fallback(
-        aten.scatter_reduce_.two,
-        self,
-        dim,
-        index,
-        src,
-        reduce=reduce,
-        include_self=include_self,
-    )
 
-    if fallback_result:
-        return fallback_result
+    if isinstance(src, Number):
+        src = full_like(self, src)
+
+    if reduce is not None:
```

Comment:
Should this check be in `use_scatter_fallback`?

Reply:
It depends on the op passed to `scatter_fallback`.

Reply:
`scatter_fallback` passes the `op_overload` argument to `use_scatter_fallback`, so I don't see why that's an issue?

Reply:
Right. Thanks for the review.